src/net_prop.cpp

```diff
@@ -1040,7 +1040,7 @@ Param initialize_param(Network &net) {
                 if (net.noise_type.compare("heteros") == 0 &&
                     j == num_layers - 1) {
                     std::tie(mw_j, Sw_j) = gaussian_param_init_ni(
-                        scale, net.gain_w[j], net.noise_gain,
+                        scale, net.out_gain, net.noise_gain, //net.gain_w[j]
```

Comment:
@bhargobdeka you dont need to introduce `out_gain` because `net.gain_w` is equivalent to `net_out_gain`. 
For example, the net layer is defined as
```
layers: [1,     7,       7,       1]
gain_w: [0,     1,       1,       1]
```
gain_w is for mean and noise_gain is for std in the heteros case

Reply:
We use the first `gain_w` for the first half i.e., mean and `noise_gain` for second half i.e. std in the case of heteros. See [this line](https://github.com/lhnguyen102/cuTAGI/blob/main/src/net_prop.cpp#L453)